pkgs/applications/window-managers/phosh/default.nix

```diff
@@ -0,0 +1,147 @@
+{ stdenv
+, pkgs
+, fetchFromGitLab
+, substituteAll
+, meson
+, ninja
+, pkgconfig
+, wrapGAppsHook
+, libhandy
+, pulseaudio
+, glib
+, gtk3
+, gnome3
+, gcr
+, pam
+, upower
+, wayland
+, dbus
+, xvfb_run
+, phoc
+, feedbackd
+, squeekboard
+, networkmanager
+, polkit
+, elogind
+, libsecret
+, git
+, writeText
+, makeWrapper
+}:
+
+let
+  gvc = fetchFromGitLab {
+    domain = "gitlab.gnome.org";
+    owner = "GNOME";
+    repo = "libgnome-volume-control";
+    rev = "ae1a34aafce7026b8c0f65a43c9192d756fe1057";
+    sha256 = "0a4qh5pgyjki904qf7qmvqz2ksxb0p8xhgl2aixfbhixn0pw6saw";
+  };
+
+  osk_desktop = writeText "sm.puri.OSK0.desktop" ''
```

Comment:
This can probably use `makeDesktopItem`.

Reply:
Is there not an upstream desktop file? Could you add an explanation comment?

Reply:
The upstream desktop file points to a stub https://source.puri.sm/Librem5/phosh/-/blob/c765ae6a5a7a8d7ace2a8e6f772bad66ad13d6b6/data/sm.puri.OSK0.desktop.in.in#L5

Reply:
I added a comment

Reply:
According to [data/meson.build](https://source.puri.sm/Librem5/phosh/-/blob/master/data/meson.build#L9), this should be installed to `desktopdir`.

Reply:
This is done here: https://github.com/NixOS/nixpkgs/pull/88767/files#diff-a7096594c619c50c00985ad768558e00R133 Is there a better way to do it I'm not aware of?

Reply:
I misunderstood. The build system produces `$out/share/applications/sm.puri.OSK0.desktop` and `$out/share/applications/sm.puri.Phosh.desktop`, with proper translations and everything.

`$out/share/applications/sm.puri.OSK0.desktop` however refers to a non-existing `$out/bin/phosh-osk-stub`. This sounds like a bug in their build system, or where's that stub supposed to come from?

Reply:
Let's carry this discussion over to https://github.com/NixOS/nixpkgs/pull/88767#discussion_r429638492.

Reply:
this is fixed with the nixos module `phosh-ui`